code/ship/shiphit.cpp

```diff
@@ -2405,6 +2420,28 @@ static void ship_do_damage(object *ship_objp, object *other_obj, const vec3d *hi
 
 	//	if ship is already dying, shorten deathroll.
 	if (shipp->flags[Ship::Ship_Flags::Dying]) {
+		if (quadrant >= 0 && !(ship_objp->flags[Object::Object_Flags::No_shields])) {
+			auto shield_impact = ConditionData {
+				ImpactCondition(shipp->shield_armor_type_idx),
+				HitType::SHIELD,
+				0.0f,
+				// we have to do this annoying thing where we reduce the shield health a bit because it turns out the last ten percent of a shield doesn't matter
+				MAX(0.0f, ship_objp->shield_quadrant[quadrant] - MAX(2.0f, 0.1f * shield_get_max_quad(ship_objp))),
+				shield_get_max_quad(ship_objp) - MAX(2.0f, 0.1f * shield_get_max_quad(ship_objp)),
+			};
+			impact_data[static_cast<std::underlying_type_t<HitType>>(HitType::SHIELD)] = &shield_impact;
```

Comment:
Same thing here.
By the time the if section closes, shield_impact will be deallocated, leaving the pointer dangling.